crates/burn-cubecl-fusion/src/engine/codegen/ir.rs

```diff
@@ -367,23 +378,26 @@ impl LocalArgsExpand {
 #[derive(CubeType, Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
 /// Unary [element wise operation](ElemwiseOp) arguments.
 pub struct UnaryFuseArgs {
-    pub input: Arg,
-    pub out: Arg,
+    pub input: FuseArg,
+    pub out: FuseArg,
 }
 
 #[derive(CubeType, Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
 /// Binary [element wise operation](ElemwiseOp) arguments.
 pub struct BinaryFuseArgs {
-    pub lhs: Arg,
-    pub rhs: Arg,
-    pub out: Arg,
+    pub lhs: FuseArg,
+    pub rhs: FuseArg,
+    pub out: FuseArg,
 }
 
 #[derive(
     CubeType, Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize,
 )]
 /// Precisions supported by [element wise operations](ElemwiseOp).
-pub enum FusePrecision {
+///
+/// The reason we have a custom type here instead of [ElemType] in for it to implement [CubeType]
+/// as well as removing some types that we don't support for fusion.
```

Comment:
Rephrasing suggestion:
```suggestion
/// This is a custom type instead of [ElemType] so it can implement [CubeType]
/// and restricts the supported types for fusion.
```